tests/test_utils/test_misc_tools.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from copy import deepcopy
+from tests.mock_classes.general_mocks import *
+from mrmustard.utils.misc_tools import *
+
+
+###############################################################################
+##########################   duck_type_checker   ##############################
```

Comment:
Intead of a comment, you can wrap this in a class and then it will also get grouped in the pytest report
```suggestion
class TestDuckTypeChecker:
    """Tests related to the ducktype checker"""
```
You'll have to add `self` as the first argument to all the test methods.